src/sessionDappService.ts

```diff
@@ -297,18 +304,18 @@ export class DappService {
     })
   }
 
-  // TODO: keep for future developments
-
-  /* public async getOutsideExecutionCall(
+  public async getOutsideExecutionCall(
     sessionRequest: OffChainSession,
     sessionAuthorizationSignature: ArraySignatureType,
+    cacheAuthorisation: boolean,
     calls: Call[],
     revision: TypedDataRevision,
     accountAddress: string,
+    provider: ProviderInterface | Provider,
```

Comment:
probably easier for them to pass the chain id, and avoid the call, as this will be called frequently latency can be an issue